### PR TITLE
Add a test suite for OS

### PR DESCRIPTION
**Pull request by Calinou**
I'm not sure how to test this class well, as some of these tests might be a bit flaky or not very useful. They still all pass on my machine.

### CODE REVIEW

File: tests/test_os.h [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+#include "core/os/os.h"
+
+#include "thirdparty/doctest/doctest.h"
+
+namespace TestOS {
+
+TEST_CASE("[OS] Environment variables") {
+	CHECK_MESSAGE(
+			OS::get_singleton()->has_environment("USER"),
+			"The USER environment variable should be present (it's defined on any desktop OS).");
```
Comment (bruvzg) [inline tests/test_os.h:43]: It's `USERPROFILE` or `HOMEPATH` on Windows.

**Comment by akien-mga** [comment]
Needs a rebase. The test seems to fail on Linux (at least on the CI).

**Comment by akien-mga** [comment]
The PR has conflicts with `master`, and somehow seems to wake up AppVeyor.

**Comment by Calinou** [comment]
This passes CI now, so it should be ready to merge :slightly_smiling_face:

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**